build-tools/packages/build-cli/src/commands/merge/branches.ts

```diff
@@ -36,146 +46,307 @@ export default class MergeBranch extends BaseCommand<typeof MergeBranch> {
 			char: "b",
 			required: true,
 		}),
-		pullRequestInfo: Flags.string({
-			description: "Pull request data",
-			char: "p",
-			multiple: true,
+		remote: Flags.string({
+			description: "",
+			char: "r",
+			default: "origin",
 		}),
 		...BaseCommand.flags,
 	};
 
+	/**
+	 * The branch that the command was run from. This is used to checkout the branch in the case of a failure.
+	 */
+	private initialBranch: string = "";
+
+	/**
+	 * A list of branches that should be deleted if the command fails.
+	 */
+	private readonly branchesToCleanup: string[] = [];
+
+	private gitRepo: Repository | undefined;
+	private remote: string | undefined;
+
 	public async run(): Promise<void> {
 		const flags = this.flags;
 
+		const prTitle: string = `Automation: ${flags.source}-${flags.target} integrate`;
 		const context = await this.getContext();
-		const gitRepo = context.gitRepo;
-		const prExists: boolean = await pullRequestExists(flags.auth, this.logger);
+		this.gitRepo ??= new Repository({ baseDir: context.gitRepo.resolvedRoot });
 
-		if (prExists) {
-			this.exit(-1);
-			this.error(`Open pull request exists`);
+		if (this.gitRepo === undefined) {
+			this.log(`gitRepo undefined: ${JSON.stringify(this.gitRepo)}`);
+			this.error("gitRepo is undefined", { exit: 1 });
 		}
 
-		const lastMergedCommit = await gitRepo.mergeBase(flags.source, flags.target);
-		this.log(
-			`${lastMergedCommit} is the last merged commit id between ${flags.source} and ${flags.target}`,
-		);
+		const repoURL = await this.gitRepo.gitClient.raw(["config", "--get", "remote.origin.url"]);
```

Comment:
There's an easier way to do this. It's not obvious. Example:

https://github.com/microsoft/FluidFramework/blob/39dbbaaebe23918339a9ef10e709eaab09f2aab3/build-tools/packages/build-cli/src/handlers/checkFunctions.ts#L208-L212

This works because we always set the COntext's originRemotePartialUrl property to microsoft/FluidFramework:

https://github.com/microsoft/FluidFramework/blob/39dbbaaebe23918339a9ef10e709eaab09f2aab3/build-tools/packages/build-cli/src/base.ts#L120

It's ok to use the old wrapper here since you don't have to add any new code to it.